lib/influx.rb

```diff
@@ -182,6 +183,44 @@ def noise_candidates(start_date = nil, end_date = nil)
       }.compact
     end
 
+    def threshold_recommendations(opts)
+      data = find_incidents(opts[:start_date], opts[:end_date],
+                            :query_select => "select count(incident_key), percentile(time_to_resolve, #{opts[:percentage]}), max(time_to_resolve)",
+                            :conditions => "group by incident_key"
+             )
+      # Firstly, don't try to provide analysis for data where we have less than 5 instances of it
+      # Also remove alerts that don't have an incident key, or haven't been resolved yet.
+      recover_within = ChronicDuration.parse(opts[:recover_within]).to_i
+      raise "Failed to parse recover within" unless recover_within > 0
+      data.reject! { |x| x['percentile'].nil? || x['percentile'].to_i > recover_within || x['count'] < opts[:more_than].to_i || x['incident_key'].nil? }
+
+      if opts[:sort_by] == 'frequency'
+        data = data.sort_by { |x| x['count'] }.reverse
+      else
+        data = data.sort_by { |x| x[opts[:sort_by]] }
+      end
+
+      data.map do |d|
+        threshold = d['percentile'] + 5
```

Comment:
This is a static 5 seconds more than the nth percentile alert.  Perhaps this should be a percentage.